### PR TITLE
NODE-2622 Delete old data

### PR DESCRIPTION
**Pull request by vsuharnikov**
A new functionality for deletion of old data. It doesn't affect the main functionality of NODE. 
Enabling this funcitonality starts a background cleanup and affects:
- REST API:
  - `GET  /addresses/balance/$addr/$confirmations` - confirmations should be <= 1000 to get a guaranteed result.
  - `GET  /addresses/balance`
  - `POST /addresses/balance`
  - `GET  /assets/$asset/distribution/$height/limit/$limit`
  - `GET  /debug/balances/history/$addr`
  - `GET  /debug/stateWaves/$height`
- Explorer tool:
  - `A`  - WAVES balance history for address
  - `AA` - Asset balance history for address

Changes:
- New setting: `waves.db.cleanup-interval` . See more information in the config;
- DB changes:
    - `Data` keys contain `AddressId` instead of `PublicKeyHash`;
    - Removed obsolete sequences: `Score`, `BlockTransactionsFee`, `BlockReward`, `WavesAmount`, `HitSource`, `BloomFilterChecksum`;
    - Added `ChangedWavesBalances` and `LastCleanupHeight` to support the new functionality;
- `Exporter` now is stack-safe;
- Improved `Explorer` compatibility with version-1.5.x and recent changes;
- Fixed resource cleanup;
- Support Apple ARM in `node-it` tests;
- Added useful comments.

### CODE REVIEW

File: node/src/main/scala/com/wavesplatform/database/Keys.scala [hunk 1]
```diff
@@ -121,8 +124,8 @@ object Keys {
   val activatedFeatures: Key[Map[Short, Int]] = Key(ActivatedFeatures, Array.emptyByteArray, readFeatureMap, writeFeatureMap)
 
   // public key hash is used here so it's possible to populate bloom filter by just scanning all the history keys
```
Comment (ivan-mashonskiy) [inline node/src/main/scala/com/wavesplatform/database/Keys.scala]: maybe delete this comment?

File: node/src/main/scala/com/wavesplatform/database/RocksDBWriter.scala [hunk 2]
```diff
@@ -680,6 +735,206 @@ class RocksDBWriter(
...
+    }
+
+  private def batchCleanupWavesBalances(fromInclusive: Height, toExclusive: Height, rw: RW): Unit = {
+    val lastUpdateAt = mutable.LongMap.empty[Height]
+
+    val updateAt     = new ArrayBuffer[(AddressId, Height)]() // AddressId -> First height of update in this range
+    val updateAtKeys = new ArrayBuffer[Key[BalanceNode]]()
+
+    val changedKeyPrefix = KeyTags.ChangedWavesBalances.prefixBytes
+    val changedFromKey   = Keys.changedWavesBalances(fromInclusive) // Doesn't matter, we need this only to parse
```
Comment (ivan-mashonskiy) [inline node/src/main/scala/com/wavesplatform/database/RocksDBWriter.scala]: what does this comment mean?
`changedFromKey` is used not only for parsing but also in `iterateOverWithSeek` as "seek" parameter

File: node/src/main/scala/com/wavesplatform/utils/RunNowExecutorService.scala [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.wavesplatform.utils
+
+import java.util
+import java.util.concurrent.{AbstractExecutorService, TimeUnit}
+
+object RunNowExecutorService extends AbstractExecutorService {
```
Comment (xrtm000) [inline node/src/main/scala/com/wavesplatform/utils/RunNowExecutorService.scala]: Is this not the same as `MoreExecutors.newDirectExecutorService()`?

File: node/src/main/scala/com/wavesplatform/database/Keys.scala (hunk 1 above)
Comment (vsuharnikov) [reply node/src/main/scala/com/wavesplatform/database/Keys.scala]: Done

File: node/src/main/scala/com/wavesplatform/database/RocksDBWriter.scala (hunk 2 above)
Comment (vsuharnikov) [reply node/src/main/scala/com/wavesplatform/database/RocksDBWriter.scala]: It means, `fromInclusive` doesn't affect the parsing result. Changed the comment to make it clear.

File: node/src/main/scala/com/wavesplatform/utils/RunNowExecutorService.scala (hunk 3 above)
Comment (vsuharnikov) [reply node/src/main/scala/com/wavesplatform/utils/RunNowExecutorService.scala]: Fixed

**Review by ivan-mashonskiy — APPROVED** [review]

**Pull request merged**